### ISSUE TITLE
Error

### ISSUE THREAD

**Issue by Edilso** [Mikadily/poloniexlendingbot#25]
Error: 'provided
![error](https://cloud.githubusercontent.com/assets/16596347/13551055/59f9900a-e30b-11e5-8b89-41af6770c1cb.jpg)
'

**Comment by Mikadily** [comment]
This error message is not very informative. But most likely you need to edit default.cfg and type in you api keys.

**Comment by jmahhh** [comment]
No, it's not the api keys, I've been using the bot for a while and get it too
sometimes. It's not a major problem though. I thought it was because I had two
instances of the bot running on different computers and they clashed once in a
while. That or a simple connectivity issue.
On Mon, Mar 7, 2016 at 3:46 AM, Mikadily <notifications@github.com> wrote:
This error message is not very informative. But most likely you need to edit
default.cfg and type in you api keys.

—
Reply to this email directly or view it on GitHub
[https://github.com/Mikadily/poloniexlendingbot/issues/25#issuecomment-192972761] .[https://github.com/notifications/beacon/ANOSPiHLtlslCDNa2TecPUiX66TWcgOkks5pqy8HgaJpZM4HqHfF.gif]

**Comment by praeluceo** [comment]
I am also unable to use the bot as it hangs at "provided" see the error below:
$ python lendingbot.py 
2016-03-08 21:14:24 Welcome to Poloniex Lending Bot
2016-03-08 21:14:24 ERROR: 'provided'
2016-03-08 21:15:25 ERROR: 'provided'
2016-03-08 21:16:25 ERROR: 'provided'

My configuration looks as so (and I believe my API key & secret are correct):
`[BOT]
#sleep between iterations, time in seconds
sleeptime = 60
#minimum daily lend rate in percent
mindailyrate = 0.1
#max rate. 2% is good choice because it's default at margin trader interface. 5% is max to be accepted by the exchange
maxdailyrate = 4.9
#The number of offers to split the available balance uniformly across the [gaptop, gapbottom] range.
spreadlend = 3
#The depth of lendbook (in percent of lendable balance) to move through before placing the first (gapbottom) and last (gaptop) offer.
#if gapbottom is set to 0, the first offer will be at the lowest possible rate. However some low value is recommended (say 10%) to skip dust offers
gapbottom = 15
gaptop = 100
#Daily lend rate threshold after which we offer lends for 60 days as opposed to 2. If set to 0 all offers will be placed for a 2 day period
sixtydaythreshold = 0.2
# AutoRenew - if set to 1 the bot will toggle the AutoRenew flag for the loans when you stop it (Ctrl+C) and clear the AutoRenew flag when started
autorenew = 0
#custom config per coin, useful when closing positions etc.
#syntax: [COIN:mindailyrate:maxactiveamount, ... COIN:mindailyrate:maxactiveamount]
#if maxactive amount is 0 - stop lending this coin. in the future you'll be able to limit amount to be lent.
#coinconfig = ["BTC:0.18:1","CLAM:0.6:1"]
`

I've played with different configuration settings for autorenew and the min/max dailyrate variables, but nothing manages to get the bot running. And yes, there is a balance in the BTC Lending wallet.

**Comment by praeluceo** [comment]
Also, my API access page says that the key I've associated with the bot has never been accessed, so the bot is failing at the API authentication step. And if I run the command with the --setAutoRenew or --clearAutoRenew options, the output looks as such:

`$ python lendingbot.py --setAutoRenew'
'2016-03-08 22:05:30 Welcome to Poloniex Lending Bot'
'2016-03-08 22:05:30 Setting AutoRenew...(Please Wait)'
'Traceback (most recent call last):'
'  File "lendingbot.py", line 241, in <module>'
'    setAutoRenew(1);'
'  File "lendingbot.py", line 223, in setAutoRenew'
'    loansCount = len(cryptoLended["provided"])'
'KeyError: 'provided''

**Comment by utochgo** [comment]
Just create new pair of API keys and use them

**Comment by keepthefaya** [comment]
ERROR: 'provided'

Same error, new api key  just generated

chmod needed ? (ubuntu)

**Comment by praeluceo** [comment]
I worked around my problem by creating a key without IP restrictions enabled. So then I reviewed what the API key list was actually saying, and at the bottom of the key, it says:
Trusted IPs:  Your current IP is 124.184.117.275

However, that does not mean that your current IP -is- a trusted IP. That just shows your current IP. I added the IP of the server I am running the script from, and when I get home I will ssh into it and change out the keys to see if that resolves the problem, but I feel it will.

I think the issue is that the "IP Access Restriction" API Keys come by default in an unusable state, due to not being configured against any IP address. The error response from the script is not useful, but the problem appears to be with an unconfigured API key, not the script or the API key itself. @keepthefaya try adding your IP address to the API key and restart the script.
